internal/git/git.go

```diff
@@ -166,9 +166,14 @@ func (r *Repo) CheckoutBranch(opts *CheckoutBranch) (string, error) {
 		args = append(args, "-b")
 	}
 	args = append(args, opts.Name)
-	if _, err := r.Git(args...); err != nil {
-		return "", err
+	res, err := r.Run(&RunOpts{
+		Args:      args,
+		ExitError: true,
```

Comment:
Instead of this, we should check `res.ExitCode` and return an error that contains the stderr from the git command. We do this in `internal/actions/push.go` (the `if res.ExitCode != 0` bit). That way the error message is a little bit less verbose.